spec/request/api/v1/sessions/create_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe 'POST api/v1/users/sign_in', type: :request do
+  let(:password) { 'password' }
+  let(:token) do
+    {
+      '70crCAAYmNP1xLkKKM09zA' =>
+      {
+        'token' => '$2a$10$mSeRnpVMaaegCpn3AhORGe5wajFhgMoBjGIrMwq4Qq2mP6f/OHu1y',
+        'expiry' => 153_574_356_4
+      }
+    }
+  end
+  let(:user) { create(:user, password: password, tokens: token) }
+
+  context 'with correct params' do
+    before do
+      params = {
+        user:
+          {
+            email: user.email,
+            password: password
+          }
+      }
+      post new_user_session_path, params: params, as: :json
+    end
+
+    it 'returns success' do
+        expect(response).to have_http_status(:success)
+    end
+
+    it 'returns the user' do
+      expect(json[:user][:id]).to eq(user.id)
+      expect(json[:user][:email]).to eq(user.email)
+      expect(json[:user][:uid]).to eq(user.uid)
+      expect(json[:user][:provider]).to eq('email')
+    end
+
+    it 'returns a valid client and access token' do
+      token = response.header['access-token']
+      client = response.header['client']
+      expect(user.reload.valid_token?(token, client)).to be_truthy
+    end
+  end
+
+  context 'with incorrect params' do
+    it 'return errors upon failure' do
+      params = {
+        user: {
+          email: user.email,
+          password: 'wrong_password!'
+        }
+      }
+      post new_user_session_path, params: params, as: :json
```

Comment:
If you set the `subject` to be assigned with this above inside the `describe` you don't need to repeat it here.